### ISSUE TITLE
[Bug] Unable to Recognize Model Name if there's an extra "/" in the path

### ISSUE THREAD

**Issue by junrushao** [mlc-ai/mlc-llm#669]
This is technically not a bug, but a weird behavior of python's `os.path` package. I encountered this error just now, actually first time in my life, and couldn't figure it out myself until @masahi told me about it. Therefore, I'd love to post this in case anyone met this issue later.

I was quantizing and compiling a Llama2 model as my daily routine, but using the command below:

```bash
python build.py  \
  --target cuda  --quantization q4f16_1   \
  --model /models/Llama-2/hf/Llama-2-7b-chat-hf/
                                             ^^^^ note this slash
```

and MLC LLM reports:

```text
Using path "/models/Llama-2/hf/Llama-2-7b-chat-hf/" for model ""
                                                              ^^ the model name is gone
```

I thought it was a breaking change or so, got frustrated but couldn't figure out why until Masa told me.

It turns out that it works if I remove the slash, i.e.:

```bash
python build.py  \
  --target cuda  --quantization q4f16_1   \
  --model /models/Llama-2/hf/Llama-2-7b-chat-hf
                                             ^^^^ slash removed
```

then MLC LLM behaves normally:

```text
Using path "/models/Llama-2/hf/Llama-2-7b-chat-hf" for model "Llama-2-7b-chat-hf"
```

We need to fix this.

**Comment by tqchen** [comment]
cc @CharlieRuan0130 for awareness, we should add it for the Es

**Comment by CharlieRuan0130** [comment]
Made a small PR; should fix the problem.

Seems to be the behavior of [os.path.basename()](https://docs.python.org/3/library/os.path.html#os.path.basename):
```python
# Without trailing `/`
os.path.basename("/models/Llama-2-7b-chat-hf")
>>> Llama-2-7b-chat-hf

# With trailing `/`
os.path.basename("/models/Llama-2-7b-chat-hf/")
>>> ""
```

This method can remove potential trailing `/`: [os.path.normpath()](https://docs.python.org/3/library/os.path.html#os.path.normpath)
```python
os.path.normpath("/models/Llama-2-7b-chat-hf/")
>>> /models/Llama-2-7b-chat-hf
```

There aren't many places in the repo where we use `os.path.basename`. For those that do, they seem to be fine except in this case.

**Comment by junrushao** [comment]
Closed via #671
